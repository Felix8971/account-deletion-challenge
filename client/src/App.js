import React from 'react'
import * as LoadState from './LoadState'
import TerminateModalFlow from './components/TerminateModalFlow';
import {
  fetchRelatedWorkspaces,
  transferOwnership,
  terminateAccount,
} from './api/index';

export default class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        _id: 'user1',
        name: 'Ross Lynch',
        email: 'ross@example.com',
      },
      loading: true,
      requiredTransferWorkspaces: [],
      deleteWorkspaces: [],
      transferableMembers: [],
      terminateAccountStatus: {},
      transferOwnershipStatus: {
        workspaceId: null,
        toUserId: null,
        ...LoadState.pending,
      },
    }
  }

  terminateAccountError = error => {
    this.setState({
      terminateAccountStatus: LoadState.handleLoadFailedWithError(error)(
        this.state.terminateAccountStatus
      ),
    })
  };

  resetTerminateAccountStatus = () => {
    this.setState({
      terminateAccountStatus: { status: null },
    })
  };

  redirectToHomepage = () => {
    window.location = 'http://www.example.com/';
  };

  render() {
    return <TerminateModalFlow 
      {...this.state} 
      fetchRelatedWorkspaces = {fetchRelatedWorkspaces(this)}
      transferOwnership = {transferOwnership(this)}
      terminateAccount =  {terminateAccount(this)}
      terminateAccountError =  {this.terminateAccountError}
      resetTerminateAccountStatus = {this.resetTerminateAccountStatus}
      redirectToHomepage =  {this.redirectToHomepage}
    />
  }
}
