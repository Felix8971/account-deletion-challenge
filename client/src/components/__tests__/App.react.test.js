import React from 'react';
import App from '../../App';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });

describe('App', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });

  it('should render correctly', () => {
    const mockFn = jest.fn();
    const tree = shallow(
      <App
        fetchRelatedWorkspaces = {mockFn}
        transferOwnership = {mockFn}
        terminateAccount =  {mockFn}
        terminateAccountError =  {mockFn}
        resetTerminateAccountStatus = {mockFn}
        redirectToHomepage =  {mockFn}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
