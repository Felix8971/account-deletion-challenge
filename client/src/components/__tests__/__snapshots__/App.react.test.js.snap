// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<TerminateModalFlow
  deleteWorkspaces={Array []}
  fetchRelatedWorkspaces={[Function]}
  loading={true}
  redirectToHomepage={[Function]}
  requiredTransferWorkspaces={Array []}
  resetTerminateAccountStatus={[Function]}
  terminateAccount={[Function]}
  terminateAccountError={[Function]}
  terminateAccountStatus={Object {}}
  transferOwnership={[Function]}
  transferOwnershipStatus={
    Object {
      "status": "pending",
      "toUserId": null,
      "workspaceId": null,
    }
  }
  transferableMembers={Array []}
  user={
    Object {
      "_id": "user1",
      "email": "ross@example.com",
      "name": "Ross Lynch",
    }
  }
/>
`;
